### ISSUE TITLE
[patch] fix for Linux kernels >= 6.6.

### ISSUE THREAD

**Issue by dreirund** [freemed/tty0tty#53]
The code needs to be updated for Linux >= 6.6:

Here is a patch that adds support for Linux >= 6.6 (needs to have the [patch for Linux >= 6.1](https://github.com/freemed/tty0tty/issues/47#issuecomment-1375882086) also be added separately):

[`linux-6.6-fixup.patch.txt`](https://github.com/freemed/tty0tty/files/13907474/linux-6.6-fixup.patch.txt):  
```
diff -rU1 tty0tty.linux-6.1-fixed/module/tty0tty.c tty0tty/module/tty0tty.c
--- tty0tty.linux-6.1-fixed/module/tty0tty.c	2024-01-11 20:13:27.739146823 +0100
+++ tty0tty/module/tty0tty.c	2024-01-11 20:14:35.892478188 +0100
@@ -212,4 +212,9 @@
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,6,0)
+static ssize_t tty0tty_write(struct tty_struct *tty, const unsigned char *buffer,
+			 size_t count)
+#else
 static int tty0tty_write(struct tty_struct *tty, const unsigned char *buffer,
 			 int count)
+#endif
 {
```  
*([Authorship goes to @ozz from Arch User Repository](https://aur.archlinux.org/packages/tty0tty-dkms-git#comment-950887).)*

[`linux-6.1-fixup.patch.txt`](https://github.com/freemed/tty0tty/files/13907478/linux-6.1-fixup.patch.txt):  
```
diff -rU1 tty0tty.org/module/tty0tty.c tty0tty/module/tty0tty.c
--- tty0tty.org/module/tty0tty.c	2024-01-11 20:08:54.922488036 +0100
+++ tty0tty/module/tty0tty.c	2024-01-11 20:10:36.829151760 +0100
@@ -290,2 +290,5 @@
 static void tty0tty_set_termios(struct tty_struct *tty,
+				#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)
+					const
+				#endif
 				struct ktermios *old_termios)
```

*(Note that tabulars are used!, copy & paste _might_ transform them to whitespace which will break applying the patches.)*

Regards!

**Comment by cschweers** [comment]
You should have checked the PR's first https://github.com/freemed/tty0tty/pull/52

**Comment by stappersg** [comment]
Solved by merge request #52. (Thanks to another merge request that reminded me that I have merge privilege for this git repository (Sorry for the late response))
